src/nbl/video/IPhysicalDevice.cpp

```diff
@@ -90,4 +90,189 @@ bool IPhysicalDevice::validateLogicalDeviceCreation(const ILogicalDevice::SCreat
     return true;
 }
 
+inline core::bitflag<asset::IImage::E_ASPECT_FLAGS> getImageAspects(asset::E_FORMAT _fmt)
+{
+    core::bitflag<asset::IImage::E_ASPECT_FLAGS> flags;
+    bool depthOrStencil = asset::isDepthOrStencilFormat(_fmt);
+    bool stencilOnly = asset::isStencilOnlyFormat(_fmt);
+    bool depthOnly = asset::isDepthOnlyFormat(_fmt);
+    if (depthOrStencil || depthOnly) flags |= asset::IImage::EAF_DEPTH_BIT;
+    if (depthOrStencil || stencilOnly) flags |= asset::IImage::EAF_STENCIL_BIT;
+    if (!depthOrStencil && !stencilOnly && !depthOnly) flags |= asset::IImage::EAF_COLOR_BIT;
+
+    return flags;
+}
+
+float getFullPrecisionValue(asset::E_FORMAT f, uint32_t channels)
+{
+    float min = std::numeric_limits<float>::infinity();
+
+    for (uint32_t i = 0; i < channels; i++)
+    {
+        min = std::min(min, asset::getFormatPrecision<float>(f, i, 0.0));
+    }
+
+    return min;
+}
```

Comment:
no, compare each channel separately, only when precision is equal or greater for EACH channel invidually you can promote.

You're being overly conservative